### DISCUSSION TITLE
ORM selection

### DISCUSSION THREAD

**Discussion by TangoYankee** [NYCPlanning/ae-zoning-api#116] [General]
# Description
Drizzle or Mikro?

**Comment by horatiorosa** [comment]
I say go with **Drizzle**. I recall working with Mikro-orm and trying to a select distinct and left join that just wouldn't work unless we used raw SQL. Drizzle had ready made methods to do that same without resorting to raw SQL.

**Reply by TylerMatteo** [reply]
Yeah I'm on board with going with Drizzle. You and @TangoYankee have the most experience using the two ORMs at this point, so I trust your opinions if you're both firmly preferring Drizzle.

To your point about needing raw SQL with Mikro-orm, it definitely seems like Drizzle is better at handling situations where you need to go slightly beyond what traditional ORMs do out of the box. We seem to be running into these situations pretty regularly (such as when using PostGIS functions), so Drizzle is the correct choice in that sense.

**Comment by TangoYankee** [comment]
We'll also want to pick an ORM before we start testing because we'll need to be mock the repos.

Here is a branch that [removes mikro orm](https://github.com/NYCPlanning/ae-zoning-api/tree/remove-mikro)

**Comment by TangoYankee** [comment]
Mikro is removed in #127
